Refit/RequestBuilderImplementation.cs

```diff
@@ -653,20 +653,33 @@ static bool DoNotConvertToQueryMap(object value)
             if (value == null)
                 return false;
 
-            var type = value.GetType().GetTypeInfo();
+            var type = value.GetType();
 
-            if (type.IsPrimitive)
+            bool ShouldReturn() => type == typeof(string) ||
```

Comment:
Does this need to be a local function? Can't you just set a boolean?

Reply:
We call the same logic twice

Reply:
👀

Reply:
The "should we keep it" logic runs once for the normal input types, then again for the `T` type if it's an enumerable.

Reply:
Ah, so a local function doesn't trap the state of `type`, and when we call it later, the value has changed? Makes sense. Ignore me then. 😂

Reply:
Yeah, works like a closure except without needing the allocation of a delegate. The `type` is captured and modified by the time it executes again.

Reply:
That's very cool.